luavm/src/lib/instructions/tables.rs

```diff
@@ -0,0 +1,81 @@
+use errors::LuaError;
+use luacompiler::bytecode::instructions::{first_arg, second_arg, third_arg};
+use luacompiler::irgen::register_map::ENV_REG;
+use Vm;
+
+/// R(1) = R(2)[R(3)]
+pub fn get_attr(vm: &mut Vm, instr: u32) -> Result<(), LuaError> {
+    let val = {
+        let arg2 = second_arg(instr) as usize;
+        let from = &vm.registers[arg2];
+        let attr = &vm.registers[third_arg(instr) as usize];
+        match attr.get_constant_index() {
+            Some(i) if arg2 == ENV_REG => vm.env_attrs[i].clone(),
```

Comment:
It feels a bit weird that nothing in the `if` arm references the left hand-side. The logic feels sort of inside-out to me.

Reply:
I don't see a reason why it should. The left hand-side is the register in which the value is copied. I am only interested to see if the value comes from `_ENV` or not. Either way, the value is copied [here](https://github.com/rbartlensky/Lua-interpreter/pull/11/files/6a8d0f7231d84c95c94f7faf325d5ccaa358c1fd#diff-ae248b349d559508173ca78a49af6e20R17) in the left hand-side. I hope I am not misunderstanding your comment.

Reply:
Oh I see, but then I would only be able to write this:
```rust
        if arg2 == ENV_REG {
            if let Some(i) = attr.get_constant_index() {
                vm.env_attrs[i].clone()
            } else {
                from.get_attr(attr)?
            }
        } else {
            from.get_attr(attr)?
        }
```

I prefer my initial solution, but this might be slightly more efficient and more readable?

Reply:
I saw [this](https://github.com/rust-lang/rfcs/pull/2497) merged, but I don't think it is part of stable. At least it doesn't work on rust playground (I made sure to check rust 2018).

Reply:
Maybe:
```rust
match attr.get_constant_index() {
   Some(i) => if arg2 == ENV_REG { vm.env_attrs[i].clone() } else { from.get_attr(attr)? }
   None => ...
}
```

Reply:
I will duplicate `from.get_attr(attr)?` this way. Is that fine?

Reply:
Your call. It feels clearer to me, but I don't feel strongly about it.